### PR TITLE
Punctualite

### PR DESCRIPTION
**Pull request by GreatBigFailure**
Adds punctualite, idea from cucumberia. 

**What is it**
Punctualite, when being actively metabolized, explodes at the turn of the hour. Drank it at 1:58? You have two minutes. 1:30? A whole half hour. It takes a lot to make a really big boom, but at high volumes it's maxes out at ~~a bit below~~ twice a welder tank explosion. 
Punctualite also has a metabolism rate of 0, so a quick sip at 12:08 is just as dangerous as 12:59

**How do I get it?**
5u of punctualite is made from 10u hyperzine, 10u welderfuel, and a catalyst of 5u zombie powder. So if you have 5u zombie powder, it's fairly easy to get a few hundred units of it.

:cl:
 * rscadd: Punctualite, a ghetto chemistry made chemical that explodes at the turn of the hour.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -10325,3 +10325,29 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 			new /obj/effect/alien/weeds(T)
 		if(!locate(/obj/effect/decal/cleanable/purpledrank) in T)
 			new /obj/effect/decal/cleanable/purpledrank(T)
+
+/datum/reagent/punchualite
+	name = "Punchualite"
+	id = PUNCHUALITE
+	description = "Nicknamed mad chemist's alarm clock. Explodes on the hour When being actively metabolized. "
```
Comment (brndd) [inline code/modules/reagents/Chemistry-Reagents.dm]: Capitalization.

Comment (GreatBigFailure) [reply code/modules/reagents/Chemistry-Reagents.dm]: That's what I get for rewriting sentences so much

**Comment by brndd** [comment]
Make it randomly play a ticking noise out of the mob (as a visible_message) if there's enough units to, say, potentially cause a breach when they explode.

**Comment by jwhitak** [comment]
~~Funny chem but isn't this a dev 2, bigger than a fuel tank? Oof.~~ I'd also recommend adding a way to remove this chem from the body in ways besides charcoal or waterpotass if possible.

~~Also, this is going to be pretty busted if abused. Drop a smoke grenade of this in a room full of things that can absorb chems, and you've suddenly got a mass of free dev 2 bombs.~~

Disregard most of that, brain derp.

**Comment by GreatBigFailure** [comment]
~~Honestly thought fuel tanks were 2. 
Either way, that's not average, that's the maximum allowed and can only be gotten at 200u. It's pretty hard to forcefeed someone 200u of something and not have them just drink charcoal. Like you're on average going to have a full half hour to cleanse this from you. 
Chem smoke wouldn't give you close to 200u. If it did we'd see medbay using smoke for medicine all the time.~~

Oh okay we'll do a strikethrough thing

**Comment by nervere** [comment]
This doesn't seem very fun or fair to be the victim of.

**Comment by pedr69** [comment]
0-19 units will make a harmless explosion
20-49 will give you ~30 brute damage
50-99 will instantly crit you with a chance of gibbing your body and head
100+ will instantly gib you and scatter your head and limbs around

is the reagent not getting removed after it explodes intentional?

**Comment by GreatBigFailure** [comment]
> 0-19 units will make a harmless explosion 20-49 will give you ~30 brute damage 50-99 will instantly crit you with a chance of gibbing your body and head 100+ will instantly gib you and scatter your head and limbs around

Thanks for the breakdown. I usually test my stuff obsessively but this one is time based, even requiring I wait the full 5 minute server startup to work correctly, so it went through a few less drafts.
Maybe I'll tone it down a bit? I mean it sounds bad on paper but the logistics of it are probably the deciding factor here. Again, on average you'll have 30 minutes to drink a bit of waterpotass or charcoal. I'll do a little nerf when I review some things, but not much? 
 
> is the reagent not getting removed after it explodes intentional?

Oh lord no. This is the first thing I've made in like 1-2 years, so the simplicity is intentional, the mistakes are predictable. Thanks for noticing that.

**Comment by pedr69** [comment]
i think its fine, even the fact it stays in after exploding - something about clocks
just add a rare visible message/noise if you have >50u since you can breach the floor at that point

**Comment by GreatBigFailure** [comment]
> i think its fine, even the fact it stays in after exploding - something about clocks just add a rare visible message/noise if you have >50u since you can breach the floor at that point

I was going to just have to empty but now that you mention it, that is a full hour before it goes off again. I'll just need to have it not immediately explode again. I'll add the ticking but I'm making it very rare. It'll calculate per tick though so if you have a few minutes it'll prooobably get seen?

**Comment by GreatBigFailure** [comment]
Alright, so:
- Didn't lower the dev from 2 to 1, but I made it 150u per dev, rather than 100u
- 0.25% chance per tick to broadcast the message "Tick.. tock" regardless of how much you have in you
- Added a thing to make sure it doesn't just keep instantly exploding every tick if it didn't kill you the first time, rather than having it clear itself from your system. You have another whole hour!
- Spelling mistake

**Comment by xXxEMBERGHOSTSQUADDIExXx** [comment]
deliberately punchualite instead of punctualite?

**Comment by GreatBigFailure** [comment]
> deliberately punchualite instead of punctualite?

Ya, double pun, because it explodes, and explosions are like punches or something.

**Comment by brndd** [comment]
That visible message won't show any kind of origin. Deliberate?

Normally visible messages are something like "[user] emits a faint ticking noise".

**Comment by GreatBigFailure** [comment]
> Deliberate?

Absolutely

**Comment by brndd** [comment]
It should probably still have three periods rather than two, and end in some kind of punctuation. t. grammar nazi

**Comment by Eneocho** [comment]
> > deliberately punchualite instead of punctualite?
> 
> Ya, double pun, because it explodes, and explosions are like punches or something.

I'd say Punctualite would work better to reinforce the whole punctuality theme; it's kinda lost on Punchualite.

**Comment by GreatBigFailure** [comment]
> I'd say Punctualite would work better to reinforce the whole punctuality theme; it's kinda lost on Punchualite.

Honestly you're probably right. Punchualite sounds a bit delicious.

**Comment by Eneocho** [comment]
Punchualite sounds like a bar drink to me. It packs a (literal?) punch.

**Comment by Blithering** [comment]
Requesting the countdown theme play on the last thirty seconds before it detonates
https://www.youtube.com/watch?v=M2dhD9zR6hk

**Comment by GreatBigFailure** [comment]
> It should probably still have three periods rather than two, and end in some kind of punctuation. t. grammar nazi

It's actually only one period not two, but then it ends in punctuation :^)

**Comment by GreatBigFailure** [comment]
Alright I changed the name to punctualite, it probably does get the idea by better

**Comment by shitmaper** [comment]
Balance suggestion:
if you are unconscious when the clock strikes 00, you don't explode, because you overslept and missed it.

**Comment by Eneocho** [comment]
add that feature but only for april fools

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -10325,3 +10325,32 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
...
+	color = "#8d8791" //rgb: 200, 165, 220
+	custom_metabolism = 0 //Wouldn't be much fun if it all got metabolized beforehand
+	var/currentHour = 0 //The hour it was introduced into the system so it doesn't blow right away
+
+/datum/reagent/punctualite/on_mob_life(var/mob/living/M)
+	if(..())
+		return 1
+	if(!currentHour)
+		currentHour = floor(world.time / 36000) + 1
+	if(prob(5) && prob(5)) //0.25% chance per tick
```
Comment (ghost) [inline code/modules/reagents/Chemistry-Reagents.dm:10350]: Tick... tock... would be more grammatically pleasing.

**Comment by ShiftyRail** [comment]
Conflicts. It's spelled "Ponctualité", btw.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 2 above)
Comment (ShiftyRail) [reply code/modules/reagents/Chemistry-Reagents.dm:10350]: I also don't like how it's a visible message. What's actually happening? Do we hear the sound?

Comment (pedr69) [reply code/modules/reagents/Chemistry-Reagents.dm:10352]: yeah uh did you test this? visible messages and playsounds from things that aren't directly on a turf don't really work

**Comment by Eneocho** [comment]
Based rubylips not suffering the fr*nch language speedmerge when????

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 2 above)
Comment (GreatBigFailure) [reply code/modules/reagents/Chemistry-Reagents.dm:10350]: I mean, it wasn't really my idea. I don't think it needs it to begin with but people wanted a visible message.

**Comment by GreatBigFailure** [comment]
> Conflicts. It's spelled "Ponctualité", btw.

I have no idea what the conflicts were or why they haunted this, it said there were none but also some. I also have no idea what you just did to fix it but I appreciate you and your saving me git-stress. Thank you.

**Comment by GreatBigFailure** [comment]
Will re-open if I feel like it

**Comment by Eneocho** [comment]
but why
it was merge-able as long as the conflicts got fixed

**Comment by GreatBigFailure** [comment]
Okay we good

**Comment by Blithering** [comment]
It lives!

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 2 above)
Comment (Kurfursten) [reply code/modules/reagents/Chemistry-Reagents.dm:10350]: This is formatted strangely as Shifty said. If it's a notice message, it should be something you notice, as in:
`You hear a ticking sound...`
or
`You hear a ticking sound coming from \the [M].`
Should this be visible to deaf players? Probably not.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 3]
```diff
@@ -10335,10 +10335,39 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
...
+		M.visible_message("<span class='notice'>Tick.. tock...</span>")
+	if((floor(world.time / 36000) + 1) > currentHour)
```
Comment (Kurfursten) [inline code/modules/reagents/Chemistry-Reagents.dm]: world.timeofday will actually just give you the formatted time
time2text(world.timeofday, "hh") will pull out just the hour in question

much less messy than this math

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 4]
```diff
@@ -10335,10 +10335,39 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
...
+		currentHour = floor(world.time / 36000) + 1
+
+/datum/reagent/punctualite/proc/punctualiteExplode(var/mob/living/H)
+	var/bigBoom = 0
+	var/medBoom = 0
+	var/litBoom = 0
+	bigBoom = min(floor(volume/150), 2) //Max breach is 2, twice a welder tank
+	medBoom = min(floor(volume/50), 4)
+	litBoom = min(floor(volume/20), 7)
+	explosion(get_turf(H), bigBoom, medBoom, litBoom, litBoom)
```
Comment (Kurfursten) [inline code/modules/reagents/Chemistry-Reagents.dm]: is this meant to have a flash radius?

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 5]
```diff
@@ -10335,10 +10335,39 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 		if(!locate(/obj/effect/decal/cleanable/purpledrank) in T)
 			new /obj/effect/decal/cleanable/purpledrank(T)
 
+/datum/reagent/punctualite
+	name = "Punctualite"
+	id = PUNCTUALITE
+	description = "Nicknamed mad chemist's alarm clock. Explodes on the hour when being actively metabolized. "
```
Comment (Kurfursten) [inline code/modules/reagents/Chemistry-Reagents.dm]: Trailing space?

Comment (Kurfursten) [reply code/modules/reagents/Chemistry-Reagents.dm]: Given that this doesn't actually metabolize, the phrasing is a little weird here. Why not "explodes on the hour while inside a living creature"?

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 3 above)
Comment (GreatBigFailure) [reply code/modules/reagents/Chemistry-Reagents.dm]: Oh thank goodness, I absolutely hated how I wrote this. Thanks.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 4 above)
Comment (GreatBigFailure) [reply code/modules/reagents/Chemistry-Reagents.dm]: Nope thanks for catching that, funnier if it's fully visible

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 2 above)
Comment (GreatBigFailure) [reply code/modules/reagents/Chemistry-Reagents.dm:10350]: I think I got it right that time

**Comment by GreatBigFailure** [comment]
Made all requested changes.

~~I broke it~~

Okay fixed it. Similar to what was there before with improved readability and I moved one conditional to a more efficient place.
Testing time based things is really annoying.

**Pull request merged**